tools/doc/allhtml.mjs

```diff
@@ -1,11 +1,10 @@
-'use strict';
-
 // Build all.html by combining the generated toc and apicontent from each
 // of the generated html files.
 
-const fs = require('fs');
+import fs from 'fs';
+import { fileURLToPath } from 'url';
 
-const source = `${__dirname}/../../out/doc/api`;
+const source = fileURLToPath(new URL('../../out/doc/api', import.meta.url));
```

Comment:
nit: fs APIs all support `URL` instead of path, that would allow "cleaner" path definition below.

https://github.com/nodejs/node/blob/0177d4cc20a6dd924fbb51b2b1f9a3b13b8d0969/tools/doc/allhtml.js#L15

```js
const toc = fs.readFileSync(new URL('./index.html', source), 'utf8');
```

It's OK if you prefer not doing that now though, it can be done in a future PR.
```suggestion
const source = new URL('../../out/doc/api', import.meta.url);
```

Reply:
I didn't want to stop on each of them to check whether it was okay to omit `fileURLToPath` but I can fix the ones you commented on.

Reply:
If that's OK, I can push a commit directly to your branch to use `URL`s instead of paths where it makes sense. GH suggestion UI is not really suited for this kind of suggestion.

Reply:
feel free to push 👍🏻